### DISCUSSION TITLE
Why no 'unet.train()' after log_validation in train_text_to_image.py

### DISCUSSION THREAD

**Discussion by wikiwen** [huggingface/diffusers#7493] [Q&A] [answered]
In diffusers/examples/text_to_image/train_text_to_image.py[line](https://github.com/huggingface/diffusers/blob/0b8e29289dd97805f778922c98a13cd0700d3ab3/examples/text_to_image/train_text_to_image.py#L1022C17-L1022C31)

As model will be "eval()" default in log_validation, why no `unet.train()` to restore after log_validation

**Comment by tolgacangoz** [comment, marked as answer]
Hi @wikiwen,
I guess `log_validation()` creates a new and independent `pipeline` by using `accelerator.unwrap_model()` and then removes it at the end. So, as I understand, it doesn't intervene previous stack's `unet`; in other words, there is no need to do `unet.train()` after `log_validation()` because there was no `unet.eval()` in the first place.

**Reply by wikiwen** [reply]
Thanks for your reply, May be
